connectors/sync_job_runner.py

```diff
@@ -216,6 +217,7 @@ async def execute(self):
             self.sync_job.log_error(e, exc_info=True)
             await self._sync_done(sync_status=JobStatus.ERROR, sync_error=e)
         finally:
+            self.sync_job.log_info("Terminating the job in error, and cleaning up")
```

Comment:
Afaik the `finally` block is always executed, regardless of whether an exception was raised or not. So this log might be misleading. We are just cleaning up here

Out of scope: We are already terminating sync in `except` clauses above this line with `self._sync_done(sync_status=JobStatus.ERROR ....` perhaps we could add logs in each of the clause (e.g. we don't log anything if job is suspended / canceled)

Reply:
omg you're totally right. I'd originally just added this as a "made it here" debugging log. I think I'll still leave something here for that purpose, but you're right, it shouldn't assume error state.

I can add extra logs too, that's easy.